src/preact/component/intersection-observer-resize.js

```diff
@@ -0,0 +1,32 @@
+import {
+  observeIntersections,
+  unobserveWithSharedInOb,
+} from '#core/dom/layout/viewport-observer';
+
+import {useCallback, useRef} from '#preact';
+
+/**
+ * Uses a shared IntersectionObserver per window instance to observe the given `ref`.
+ *
+ * @param {function(IntersectionObserverEntry)} callback
+ * @return {function(Element)}
+ */
+export function useIntersectionObserver(callback) {
+  const nodeRef = useRef(null);
+  const refCb = useCallback(
+    (node) => {
+      const prevNode = nodeRef.current;
+      nodeRef.current = node;
+      if (prevNode) {
+        unobserveWithSharedInOb(prevNode);
```

Comment:
@caroqliu: good point here. Ideally we have an `unobserveIntersections` API that only removes the single callback instead of all observers! Deserves its own unit test as well.

Reply:
Approving with the understanding that this should be addressed ASAP but not necessarily in this PR (given it is existing behavior)

Reply:
Ack. Added todo.